modules/consensus/validtransaction.go

```diff
@@ -120,11 +120,37 @@ func validStorageProofs(tx *bolt.Tx, t types.Transaction) error {
 
 		// COMPATv0.4.0
 		//
-		// Fixing the padding situation resulted in a hardfork. The below code
-		// will stop the hardfork from triggering before block 21,000.
+		// HARDFORK 21,000
+		//
+		// Originally, the code used the entire segment to verify the
+		// correctness of the storage proof. This made the code incompatible
+		// with data sizes that did not fill an entire segment.
+		//
+		// This was patched with a hardfork in block 21,000. Ironically, modern
+		// implementations use padding to complete the segments. The
+		// introduction of this hardfork made it possible to use uneven file
+		// sizes such that the last segment was incomplete, but introduced a
+		// new bug where a full final segment made a storage proof impossible
+		// if the final segment is selected.
+		//
+		// Ironically, we now only use complete segments in our storage proof.
+		// Had we never made the original hardfork, the current implementation
+		// would be bug free. Unfortunately, a second hardfork was necessary to
+		// fix the new bug.
```

Comment:
this is unnecessarily verbose and you use "ironically" twice. Just explain the prior behavior and why it was wrong.